internal/ldconfig/ldconfig.go

```diff
@@ -131,32 +138,36 @@ func (l *Ldconfig) UpdateLDCache() error {
 		return fmt.Errorf("failed to update ld.so.conf.d: %w", err)
 	}
 
+	// In most cases, the hook will be executing a host ldconfig that may be configured widely
+	// differently from what the container image expects. The common case is Debian vs non-Debian.
+	// But there are also hosts that configure ldconfig to search in a glibc prefix
+	// (e.g. /usr/lib/glibc). To avoid all these cases, append the container's expected system
+	// search paths to the top-level ld.so.conf. This will ensure they get scanned but won't
+	// materially change the scan order.
+	if err := appendSystemSearchPathsToLdsoconf(topLevelLdsoconfFilePath, l.getSystemSearchPaths()...); err != nil {
```

Comment:
My one question here would be whether we want to use a drop in file. I suppose the issue there is that we're not guaranteed that the file we create will be lexographically last in the list of included files.